dashboard/app/controllers/aichat_controller.rb

```diff
@@ -84,16 +119,28 @@ def chat_completion
     {messages: messages}
   end
 
-  private def has_required_params?
-    begin
-      params.require([:newMessage, :aichatModelCustomizations, :aichatContext])
-    rescue ActionController::ParameterMissing
-      return false
+  private def has_required_params?(endpoint)
```

Comment:
nit: would it be simpler to just check the params inside each function directly since we're checking a different set of params for each? is there much value in having the shared function?

Reply:
Oh yeah - I think I initially had the params being very similar but they've diverged so checking directly sounds good. Don't have a preference either way.

Reply:
I think I'll keep the helper function for `chat_completion` bc a little more complex.